STYLEGUIDE.md

```diff
@@ -46,17 +46,19 @@ import (
 	"context"
 	"fmt"
       // above are standard library packages
+	appsv1 "k8s.io/api/apps/v1"
 	"k8s.io/client-go/tools/record"
 	"sigs.k8s.io/kustomize/k8sdeps/kunstruct"
 	"sigs.k8s.io/kustomize/pkg/target"
 	ktypes "sigs.k8s.io/kustomize/pkg/types"
       // above are k8s library packages
 	"github.com/kudobuilder/kudo/pkg/util/health"
 	"github.com/kudobuilder/kudo/pkg/util/template"
-	appsv1 "k8s.io/api/apps/v1"
 )
 ```
 
+Executing `make import` should fix the imports for you.
```

Comment:
But the above "import" section doesn't correspond to `goimports` rules. It's outdated, if this isn't out-of-scope for this PR, let's fix it here as well.

Reply:
OK, turns out that what goimports does is a little tricky. I experimented a bit with the following file, and I think I captured the most important stuff. PTAL @nfnt 

```
package main

import (
	"context"
	"fmt"

	"github.com/onsi/gomega"
	"github.com/sirupsen/logrus"

	appsv1 "k8s.io/api/apps/v1"
	"k8s.io/client-go/tools/record"
	"sigs.k8s.io/kustomize/k8sdeps/kunstruct"
	"sigs.k8s.io/kustomize/pkg/target"
	ktypes "sigs.k8s.io/kustomize/pkg/types"

	"github.com/kudobuilder/kudo/pkg/util/kudo"
	"github.com/kudobuilder/kudo/pkg/version"
)

func main() {
	context.Background()
	fmt.Printf("a %v", appsv1.ControllerRevision{})
	record.NewBroadcaster()
	kunstruct.NewKunstructuredFactoryImpl()
	target.MakeEmptyAccumulator()
	ktypes.NewGenerationBehavior("a")
	kudo.String("a")
	gomega.BeClosed()
	version.FromGithubVersion()
	logrus.Error()
}
```

Reply:
The above is because `goimports -w` produces unreliable output that cannot be trusted. At least it didn't remove any packages, breaking the compilation, like it did when I tried it. IMO, we should remove the `make imports` step. cc @kensipe

Reply:
Hm, really? Can you reproduce this? I think every time I had an issue with `goimports` it was actually my fault in the end...

Reply:
Deleted my last comment as this isn't about `goimports`. While the above example captures what's described in the current import guidelines and is something that passes the linter, this isn't what `make imports` _would_ create. That output will only have 3 sections: Go packages, followed by third-party packages, followed by `kudobuilder/kudo` packages. So this should be
```suggestion
import (
	"context"
	"fmt"

	"github.com/onsi/gomega"
	"github.com/sirupsen/logrus"
	appsv1 "k8s.io/api/apps/v1"
	"k8s.io/client-go/tools/record"
	"sigs.k8s.io/kustomize/k8sdeps/kunstruct"
	"sigs.k8s.io/kustomize/pkg/target"
	ktypes "sigs.k8s.io/kustomize/pkg/types"

	"github.com/kudobuilder/kudo/pkg/util/kudo"
	"github.com/kudobuilder/kudo/pkg/version"
)
```
and the "import" section updated accordingly.

Reply:
> this isn't what make imports would create

This statement is not universally true. As described in the text I added below this example, the output depends on what the input is. So if one introduces a blank line after the logrus import, as required by the style guide rule above the example, then `goimports` (as invoked by our `make imports`) will keep it as is. I tested this.

Reply:
You're right! I didn't know, always test with all blank lines removed and checked what gets changed by `goimports`.
Hmm, while that would allow us to follow our imports guidelines, this isn't what's in our code currently. While adding the `make imports` target, all files have been updated to have the 3 sections I mentioned above. I'd argue to keep this at 3 sections, as it will be much easier to enforce: When removing all blank lines from a files import section, `make imports` will always sort them into these 3 sections. cc @kensipe

Reply:
And I think alphabetical sorting within the middle section should give us 90% of what we want, since `k8s.io/...` and `sigs.k8s.io/...` will still be kept close together.